src/App.js

```diff
@@ -1,25 +1,48 @@
-import logo from './logo.svg';
+import { useState, useEffect } from 'react';
 import './App.css';
+import destinations from "./destinations.json";
+import DestinationBox from './components/DestinationBox';
 
 function App() {
+  console.log(destinations);
+  const [locations, setLocations] = useState(destinations.destinations || []);
+
+  const removeDestination = (index) => {
+    let newLocations = [...locations];
+    newLocations.splice(index, 1);
+    setLocations(newLocations)
+  }
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      {/* <ul>
+      {locations.map((location, index) => (
+        <li key={index}>
+          {location.name} {" "} 
+          <img src={location.image} width={150} alt={location.name} />
+        </li>
+      ))}
+      </ul> */}
+      <h1>
+        Explore Destinations
+      </h1>
+      <div>
+        {locations.map((location, index) => (
+          <DestinationBox key={index} destination={location} onDelete={(index) => removeDestination(index)} />
```

Comment:
avoid using the index as a key, the deletion is not working
in this case it's because you're passing index in onDelete as an argument:
` onDelete={(index) => removeDestination(index)}`
should be
`onDelete={() => removeDestination(index)}`
but even though this will work cause it's a simple object, I'd recommend not to use the index as a key, use an id or the name instead, you can find the index with findIndex built in function
 
from the slides:
"As a last resort, you can pass an item’s index in the array as a key. This can work well if the items are never reordered; otherwise it can negatively impact performance and may cause issues with component state
If items are going to be reordered it may be preferable to use the item value itself as the key rather than indexes"

Reply:
Apuntado!! Estaba revisando el trabajo que hicimos con Rana por la mañana y habíamos escrito una función similar - esto fue con el índice como argumento, así que seguí esa lógica 🤷‍♀️  Aun así, prometo que me funciona con él dentro 🥲 
<img width="468" alt="Screenshot 2024-10-21 at 21 42 31" src="https://github.com/user-attachments/assets/5770e9a5-9153-487c-b6fb-93653bed2580">

Reply:
The problem is that you're shadowing the index variable here(I'd like to see the whole code in that other excercise9. You're already using index in the .map() callback (for the array index), but when you pass onDelete={(index) => removeDestination(index)}, you're overwriting the index argument in the onDelete callback function. This means that index in onDelete **might** not refer to the correct value anymore.

Fix: You should avoid shadowing the variable by either renaming it or passing it directly from the parent scope:
`onDelete={() => removeDestination(index)}`

I'll show you tomorrow